review: sample_geometric_exp_slow

Shoeboxam opened the pull request:
Towards #2069

Shoeboxam commented:
* **#2118** <a href="https://app.graphite.dev/github/pr/opendp/opendp/2118?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Shoeboxam and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

michaelghay commented:
@Shoeboxam attached is my proof review.
[sample_geometric_exp_slow.pdf](https://github.com/user-attachments/files/18215483/sample_geometric_exp_slow.pdf)

Shoeboxam commented:
@michaelghay 

1. Those commit hashes ostensibly achieve what we are setting out to do, but are incredibly prone to change. Updated.
2. I notice a comment on the `#` in the code. This is where a latex \label is, linked to from the proof. I could do more to hide the hash, but it's also kind of nice knowing there's a label there.
3. The link is right, but the Rust documentation failed to build after OpenDP 0.12's release yesterday. It will fix itself once #2212 is merged and 0.12.1 is released.
4. The correct output distribution is $(1 - \exp(-x)) \exp(-x)^{k}$, and indeed negating the bools gives us this distribution. Good catch! 

I fixed the above, added distribution definitions, and cleaned up some of the formatting and structure in the "respond to review commit". 

Thanks for taking the time to look at this!

michaelghay reviewed (approved):
I reviewed the revisions to the proof and they look good to me.  Thanks @Shoeboxam!

tholop reviewed (approved):
Looks good to me!
